source/docs/hardware/sensors/mb1013-hrlv-maxsonar-ez-ultrasonic-sensor.rst

```diff
@@ -0,0 +1,84 @@
+MB1013 HRLV-MaxSonar-EZ Ultrasonic Sensor
+=========================================
+
+.. image:: images/mb1013-hrlv-maxsonar-ez-ultrasonic-sensor/sensor.png
+
+Below we have a tutorial to help you get started with the `MaxBotix MB1013 ultrasonic sensor <https://www.maxbotix.com/Ultrasonic_Sensors/MB1013.htm>`__! This will hopefully give you a basic understanding of how ultrasonic sensors work and will help you set up your ultrasonic sensor to read analog voltage.
+
+*Take a look at this sensor’s `data sheet <https://www.maxbotix.com/documents/HRLV-MaxSonar-EZ_Datasheet.pdf>`__ for more information on the sensor’s specs!*
+
+How Ultrasonics Work
+--------------------
+
+Using sound, the ultrasonic rangefinder measures distance for you. The diagram below shows how the sensor sends and receives the sound waves. It measures the distance using the time it takes for the sound wave to leave, reflect off of a surface or object, and travel back to the sensor.
+
+.. image:: images/mb1013-hrlv-maxsonar-ez-ultrasonic-sensor/how-ultrasonics-work.png
+
+Using Your Ultrasonic Sensor in your Project
+--------------------------------------------
+
+1. Hook Up Controller
+^^^^^^^^^^^^^^^^^^^^^
+We used an Arduino in this example, but you can use another controller and program of your choice.
+
+.. image:: images/mb1013-hrlv-maxsonar-ez-ultrasonic-sensor/hook-up-controller.png
+
+2. Install Software
+^^^^^^^^^^^^^^^^^^^
+
+Install Arduino Sketch coding software onto your PC. This is where you type the code you want to compile and send to the Arduino board.
+
+3. Set Up Your Sensor with Arduino
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Plug your Arduino into the USB cable and into your computer. Once you upload Arduino, you can then compile and activate the code.
+
+4. Compile and Run Code
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The code below will allow you to read distance in centimeters. Compile and run this code to obtain real-time distance measurements to the closest object. (Please note: this code is not only for Arduino and will run on most controllers)
```

Comment:
```suggestion
The code below will allow you to read distance in centimeters. Compile and run this code to obtain real-time distance measurements to the closest object. (Please note: this code is not only for Arduino and will run on most controllers. Also be aware that there is no Java version of the below snippet, due to the Java Virtual mMachine being difficult to run on an Arduino.)
```

There's also issues with having to use the JNI, but this is a simple enough answer